### DISCUSSION TITLE
tcc: error: undefined symbol '_ftelli64'

### DISCUSSION THREAD

**Discussion by qaisyamir** [raysan5/raylib#2268] [Q&A]
I've been having trouble to compile a few coding from the raylib 'example' folder.
It always show this error:

tcc: error: undefined symbol '_ftelli64'

I even compile in it the original folder of raylib when we first installing the raylib.

One of the program I try is the models_cubicmap.c
![image](https://user-images.githubusercontent.com/91795107/148374200-77503533-76c9-45ae-9bef-51a1d360cb9c.png)

**Comment by Bioship** [comment]
msvcrt.dll doesn't export '_ftelli64' and tcc doesn't implement it

* the reason this isn't a problem on mingw64-gcc is they implement it statically using functions that actually are exported by msvcrt.dll

'_ftelli64' is used by 'cgltf' which is included in 'rmodels.c'

you can replace the include with something like this:
```
#if defined(_WIN32) && defined(__TINYC__)
  #ifdef __int64
    #undef __int64
  #endif
  #define __int64 long
  #define _ftelli64 ftell
#endif

#define CGLTF_IMPLEMENTATION
#include "../1-cgltf/cgltf.h"

#if defined(_WIN32) && defined(__TINYC__)
  #undef __int64
  #define __int64 long long
  #undef _ftelli64
#endif
```
^ in which case your file size will be limited to a 32-bit integer, but it compiles

**Comment by spytheman** [comment]
You can use [_fileno](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/fileno?view=msvc-170) and [_telli64](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/tell-telli64?view=msvc-170), with tcc.

_fileno for getting a file descriptor (an int) from your `FILE*` stream, and then use that descriptor with `_telli64` to get a 64 bit position.

Something like this:
```c
#if defined(_WIN32) && defined(__TINYC__)
  #define _ftelli64(stream) (_telli64(_fileno(stream)))
#endif
```

**Comment by Peter0x44** [comment]
mingw-w64 headers do expose these functions, I managed to get it working by copying headers from a mingw-w64 gcc installation over to tcc. Somewhat of a hack but I did not see a better way.
